artifactory_test.go

```diff
@@ -820,7 +822,10 @@ func TestArtifactoryClientCert(t *testing.T) {
 	// Without loading a valid client certificate, we expect all actions to fail due to error: "tls: bad certificate"
 	searchCmd := generic.NewSearchCommand()
 	searchCmd.SetRtDetails(artifactoryDetails).SetSpec(fileSpec)
-	err = searchCmd.Search()
+	reader, err := searchCmd.Search()
+	if reader != nil {
```

Comment:
The reader returned by the ```searchCmd.Search()``` function can be nil? If so, why aren't we doing nil checks it in order places before closing it? (see other usages in this file, and possible in other places).

Reply:
The test supposed to fail and validate the output error, if something goes wrong, we should close the reader and fail for error validation.